sam_view.c

```diff
@@ -286,6 +361,31 @@ static inline void change_flag(bam1_t *b, samview_settings_t *settings)
         b->core.flag &= ~settings->remove_flag;
 }
 
+int parse_aux_list(auxhash_t *h, char *optarg) {
+    if (!*h)
+        *h = kh_init(aux_exists);
+
+    while (strlen(optarg) >= 2) {
+        int x = optarg[0]<<8 | optarg[1];
+        int ret = 0;
+        kh_put(aux_exists, *h, x, &ret);
```

Comment:
Ideally `ret` should be checked here, although you probably won't get far if you're out of memory this early...

Reply:
Added, although I'm not bothering to report an error.  Frankly if your system is so stressed out it can't even parse command line arguments correctly then I think you have bigger problems than worrying about the lack of a nice message! (Plus it'll be *really* obvious in a myriad of other failures.)